src/lib.rs

```diff
@@ -2,40 +2,72 @@
 
 #![cfg_attr(not(test), no_std)]
 
-const RTC_DR: usize = 0x00; //Data Register
-const RTC_LR: usize = 0x08; //Load Register
+use core::ptr::{addr_of, addr_of_mut};
 
-/// The System Real Time Clock structure for aarch64 based on PL031.
-pub struct Rtc {
-    base_address: usize,
+#[repr(C, align(4))]
+struct Registers {
+    /// Data register
+    dr: u32,
+    /// Match register
+    mr: u32,
+    /// Load register
+    lr: u32,
+    /// Control register
+    cr: u8,
+    _reserved0: [u8; 3],
+    /// Interrupt Mask Set or Clear register
+    imsc: u8,
+    _reserved1: [u8; 3],
+    /// Raw Interrupt Status
+    ris: u8,
+    _reserved2: [u8; 3],
+    /// Masked Interrupt Status
+    mis: u8,
+    _reserved3: [u8; 3],
+    /// Interrupt Clear Register
+    icr: u8,
+    _reserved4: [u8; 3],
 }
 
-impl Rtc {
-    unsafe fn read(&self, reg: usize) -> u32 {
-        core::ptr::read_volatile((self.base_address + reg) as *const u32)
-    }
-
-    unsafe fn write(&self, reg: usize, value: u32) {
-        core::ptr::write_volatile((self.base_address + reg) as *mut u32, value);
-    }
+/// The System Real Time Clock structure for aarch64 based on PL031.
+pub struct Rtc {
+    registers: *mut Registers,
 }
 
 impl Rtc {
-    /// Construct a new PL031 RTC structure.
+    /// Constructs a new instance of the RTC driver for a PL031 device at the given base address.
+    ///
+    /// The base address may be obtained from the device tree.
     ///
-    /// `base_addr` represents the device address
-    ///  (which can be obtained from the device tree).
-    pub fn new(base_address: usize) -> Self {
-        Rtc { base_address }
+    /// # Safety
+    ///
+    /// The given base address must point to the MMIO control registers of a PL031 device, which
+    /// must be mapped into the address space of the process as device memory and not have any other
+    /// aliases. It must be aligned to a 4 byte boundary.
+    pub unsafe fn new(base_address: *mut u32) -> Self {
```

Comment:
Add `unsafe` to the example in `README`.

Reply:
Done.